### ISSUE TITLE
Исправление неточностей

### ISSUE THREAD

**Issue by Skorik99** [iyphc/ecg-cnn-comparator#7]
1. Строки [179](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L179)-[181](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L181) повторяются. При этом features_num сохраняется в json формате. Явный баг
2. В функции [load_raw_data](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L69) неявно пропускаются те файлы, которые не удалось подгрузить в `try` блоке. При этом [Y](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L118) никак это не учитывает: если несколько файлов сбросится, то файлы будут несостыкованы, поскольку [итерирование](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L42) идёт по индексам, явно подразумевая их полное соответствие
3. **Дискуссионный вопрос** Насколько ок [нормализовывать](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/data/preprocess.py#L125) метаданные вдоль всех признаков? Пол -- бинарный признак, а остальные имеют разный числовой масштаб. В результате значение признака искажается.
4. В [train.py](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/train.py#L81) первым аргументом передаётся `train`, хотя функция ожидает `model`
5. Какая-то неразбериха с [инстанцированием модели](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/train.py#L21C11-L21C27): 2 условия на то, что модель is not None, при этом она же инциализируется + elif никак не связан с if.
6. Путаница с dataset и dataloaders: в функции `train_model` аргументы названы как `train_set` (хотя по факту являются даталоадерами), при этом в функции есть аргумент batch_size, который никак не используется, поскольку должен передаваться в функцию `get_dataloaders`
7. Как мы адресуем к несбалансированности классов? Зачем используем валидацию? Почему [здесь](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/evaluation.py#L96) хардкодим значение порога 0.5? 
8. Как работает roc-auc? Намек на то, что [здесь](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/evaluation.py#L71) баг.
9. В base_scores подсчитали матрицу смежности для всех патологий. С этой матрицей можно делать макро, микро и взвешенные усреднения. Почитай про них поподробнее и реализуй более подробный подсчет метрик
10. Дизайн архитектуры base_model, кажется, имеет смысл дополнить модулями batch_norm, dropout итп.
11. HandCraftedModel выстроена некорректно: BaseModel выдаёт логиты классов, к которым приклеиваются метаданные. В этом нет никакого смысла. Логиты классов уже несут смысл вероятности, которые затем переводятся в фичи, а затем снова в классы. Нужно, чтобы фичи метаданных приклеивались к фичам (с предпоследнего слоя) BaseModel.
12. Строка [26](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/cnn_handcrafted.py#L26C29-L26C45) выдаст ошибку, поскольку атрибут `self.out_classes` нигде не инициализирован. 
13. Какой смысл [переинициализировать](https://github.com/iyphc/ecg-cnn-comparator/blob/main/src/models/cnn_handcrafted.py#L10) `self.device`, если он уже инициализируется в базовом классе.
